shunt/shell_command_helper.py

```diff
@@ -0,0 +1,94 @@
+"""Module to help run shell commands and retrieve output"""
+
+from __future__ import absolute_import
+import subprocess
+import multiprocessing
+
+
+class ShellCommandHelper():
+    """Hosts methods to run shell commands and retrieve output"""
+
+    TEN_MIN_SEC = 10 * 60
+
+    def __init__(self):
+        pass
+
+    def _run_process_command(self, command, capture=False):
+        """
+        Args:
+            command: Takes an str line or list[str] to be run in shell
+            capture: Setting to True captures stdout, stderr and return code
+        returns:
+            Popen object
+        """
+        command_list = command.split() if isinstance(command, str) else command
+        pipeout = subprocess.PIPE if capture else None
+        return subprocess.Popen(command_list, stdout=pipeout, stderr=pipeout)
+
+    def _reap_process_command(self, process):
+        """
+        Args:
+            process: Popen object to reap
+        returns:
+            return code of command, stdout, stderr
+        """
+        process.wait(timeout=self.TEN_MIN_SEC)
+        stdout, stderr = process.communicate()
+        strout = str(stdout, 'utf-8') if stdout else None
+        strerr = str(stderr, 'utf-8') if stderr else None
+        return process.returncode, strout, strerr
+
+    # pylint: disable=too-many-arguments
+    def run_cmd(self, cmd, arglist=None, strict=True,
+                capture=False, docker_container=None, detach=False):
+        """
+        Args:
+            cmd: command to be executed
+            arglist: Additional arguments to add to command
+            strict: Raises exception if command execution fails
+            capture: Prints out and captures stdout and stderr output of command
+            docker_container: Set if command needs to be run within a docker container
+        returns:
+            return code of command, stdout, stderr
```

Comment:
Are the return values used anywhere? It looks like everywhere that calls run_cmd discard the return values.

Reply:
None so far, but it could be used for debugging/checking if the command went through. Bound to be useful in future dev work. It is derived from the ShellCommandHelper in Forch.